src/main/java/seedu/ccacommander/model/CommandHistory.java

```diff
@@ -0,0 +1,90 @@
+package seedu.ccacommander.model;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * The facilitator of commands and functions related to recalling previously entered commands.
+ */
+public class CommandHistory {
+    private List<String> commandHistoryList;
+    private int currentCommandPointer;
+
+    /**
+     * Constructs a new instance of CommandHistory.
+     */
+    public CommandHistory() {
+        this.commandHistoryList = new ArrayList<>();
+        this.currentCommandPointer = 0;
+    }
+
+    /**
+     * Adds the command into {@code commandHistoryList} and sets the pointer to the new command only if the command
+     * is not the latest command added.
+     *
+     * @param command
+     */
+    public void addCommand(String command) {
+        requireNonNull(command);
+        if (this.commandHistoryList.size() == 0
+                || !this.commandHistoryList.get(this.commandHistoryList.size() - 1).equals(command)) {
+            this.commandHistoryList.add(command);
+            resetPointer();
+        }
+    }
+
+    /**
+     * @return true if there is a next command.
+     */
+    public boolean hasNextCommand() {
+        boolean hasCommands = this.commandHistoryList.size() > 0;
+        boolean pointerHasNext = this.currentCommandPointer < this.commandHistoryList.size() - 1;
+        return pointerHasNext && hasCommands;
+
+    }
+
+    /**
+     * @return true if there is a previous command.
+     */
+    public boolean hasPreviousCommand() {
+        boolean hasCommands = this.commandHistoryList.size() > 0;
+        boolean pointerHasPrevious = this.currentCommandPointer > 0;
+        return pointerHasPrevious && hasCommands;
+    }
+
+    /**
+     * @return next command in the {@code commandHistoryList} based on the {@code currentCommandPointer}
+     */
+    public String getNextCommand() {
+        this.currentCommandPointer++;
+        String nextCommand = this.commandHistoryList.get(this.currentCommandPointer);
+        return nextCommand;
+    }
+
+    /**
+     * @return previous command in the {@code commandHistoryList} based on the {@code currentCommandPointer}
+     */
+    public String getPreviousCommand() {
+        this.currentCommandPointer--;
```

Comment:
Same for this line too. Will `currentCommandPointer` decrease till it becomes negative?

Reply:
I have added an assertion statement that asserts currentCommandPointer must be > 0 for this case